dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcCompletenessTableManager.java

```diff
@@ -249,7 +249,9 @@ private List<AnalyticsTableColumn> getColumns() {
         "concat(ds.uid,'-',ps.iso,'-',ous.organisationunituid,'-',acs.categoryoptioncombouid) as id ";
     String diffInSeconds = sqlBuilder.differenceInSeconds("cdr.date", "ps.enddate");
     String timelyDateDiff = diffInSeconds + " / (" + SECONDS_PER_DAY + ")";
-    String timelyAlias = "((" + timelyDateDiff + ") <= ds.timelydays) as timely";
+    // Since the end date is reported with a time of 00:00:00, we add 1 day to the timely days
+    // so that the calculations starts at 00:00:00 of the day after the end date.
```

Comment:
_starts_? or _ends_? IDK this logic, maybe you want everything after the end date then starts is correct, just double checking.

Reply:
yeah,  maybe the comment is more confusing than it needs to be.  The issue here is that ps.enddate is always going to be the last day of the period, starting at midnight. However, for the purposes of calculating timeliness, we should only "start the clock" at midnight of the day **after** the period end date.